airflow/contrib/operators/dataproc_operator.py

```diff
@@ -170,6 +172,7 @@ def __init__(self,
                  idle_delete_ttl=None,
                  auto_delete_time=None,
                  auto_delete_ttl=None,
+                 single_node=False,
```

Comment:
Instead of an explicit parameter would it be better to infer that a single node cluster is being requested if zero workers are requested and allow that?

Reply:
That sounds less error prone, so long as we document it, that works for me (but I have no context on this PR or DataProc)

Reply:
I based the behavior on how the `gcloud dataproc clusters create` command works. 

While the cluster is in a single node configuration, user has to keep in mind that resouces avaiable to the spark are not the same as in case of regular workers - single node acts both as a worker and as the master, so part of the resources is consumed by master process. 

And last but not least, I prefer to follow the rule ["Explicit is better than implicit"](https://www.python.org/dev/peps/pep-0020/).

Reply:
Right now if feels a bit redundant te me. In this case you would need to set both the number of workers, and the `single_node` to True. Since the number of workers is a required argument, I'd say that setting `num_workers` would already suggest that you want a single node cluster. I'd set the `single_node = num_workers == 0` to still be explicit in the code, but having a less redundant interface to the developer.

Reply:
Agree on your suggestions - please review the updated code

Reply:
Love the suggestion and the change.